onchain/src/social/account.cairo

```diff
@@ -48,6 +69,69 @@ pub mod SocialAccount {
         ) { // TODO: implement handle transfer logic
         }
     }
+
+    #[external(v0)]
+    impl SRC6Impl of interface::ISRC6<ContractState> {
+        fn __execute__(self: @ContractState, mut calls: Array<Call>) -> Array<Span<felt252>> {
+            let sender = get_caller_address();
+            assert(sender.is_zero(), 'Account: invalid caller');
+
+            // Check tx version
+            let tx_info = get_tx_info().unbox();
+            let version = tx_info.version;
+            if version != TRANSACTION_VERSION {
+                assert(version == QUERY_VERSION, 'Account: invalid tx version');
+            }
+
+            let mut res = ArrayTrait::new();
+            loop {
+                match calls.pop_front() {
+                    Option::Some(call) => {
+                        let Call{to, selector, calldata } = call;
+                        let _res = starknet::call_contract_syscall(to, selector, calldata.span())
+                            .unwrap();
+                        res.append(_res);
+                    },
+                    Option::None(_) => { break (); },
+                };
+            };
+            res
+        //  _execute_calls(calls)
+        }
+
+        fn __validate__(self: @ContractState, mut calls: Array<Call>) -> felt252 {
+            // self.validate_transaction()
+            let tx_info = get_tx_info().unbox();
+            let tx_hash = tx_info.transaction_hash;
+            let signature = tx_info.signature;
+            if self.is_valid_signature(tx_hash, signature) != starknet::VALIDATED {
```

Comment:
why not just return `self.is_valid_signature(tx_hash, signature)`?

Reply:
i see, so whatever is_valid_signature return is what __validate__ will return, yaah both of them have the same type of felt252